server.js

```diff
@@ -1,26 +1,110 @@
-import express from 'express'
-import bodyParser from 'body-parser'
-import cors from 'cors'
-import mongoose from 'mongoose'
+import express from "express"
+import bodyParser from "body-parser"
+import cors from "cors"
+import mongoose from "mongoose"
 
+// CODE FOR DATABASE
 const mongoUrl = process.env.MONGO_URL || "mongodb://localhost/happyThoughts"
 mongoose.connect(mongoUrl, { useNewUrlParser: true, useUnifiedTopology: true })
 mongoose.Promise = Promise
 
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+// THOUGHTS MODEL
+const Thought = mongoose.model("Thought", {
+  message: {
+    type: String,
+    required: true,
+    minlength: 5,
+    maxlength: 140,
+  },
+  hearts: {
+    type: Number,
+    default: 0,
+  },
+  name: {
+    type: String,
+    default: "Anonymous",
+  },
+  createdAt: {
+    type: Date,
+    default: () => new Date(), 
+  },
+})
+
+// SERVER
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
+// MIDDLEWARES
 app.use(cors())
 app.use(bodyParser.json())
+const listEndpoints = require("express-list-endpoints")
+
+// ERROR HANDLING WHEN DATABASE IS DOWN OR OUT OF REACH
+app.use((req, res, next) => {
+  if (mongoose.connection.readyState === 1) {
+    next()
+  } else {
+    res.status(503).json({ error: "Service unavailable" })
+  }
+})
+
+// ROOT ENDPOINT
+app.get("/", (req, res) => {
+  res.send(listEndpoints(app))
+})
```

Comment:
It will be perfect to add some error handling codes here too. For example, 
if (!res) {
    res
      .status(404)
      .send({ error: 'Oops! Something goes wrong. Try again later!' });
  }
  res.send(endPointList(app));

Reply:
Thanks I will add that :)